### PR TITLE
Rules tests rewritten

### LINKED ISSUE #80: Backend: Improve tests
I've realized many backend tests have more dependencies than needed (`effects_test` really does NOT need to call `create-game` for anything) as well as some that can simply be greatly improved in style and clarity.

It would help a lot to have another pass and maybe establish concrete guidelines for unit tests.

### PR DESCRIPTION
**Pull request by kenan-rhoton**
Relates to #80.

The following criteria are being followed:
- Minimize the amount of dependencies in each test file
- Have each test file be about a single unit when possible
- Explode big test files into their own folder

### CODE REVIEW

**Comment by kenan-rhoton** [comment]
Api tests refactor is still missing.

File: backend/test/rules/new_game_test.clj [hunk 1]
```diff
@@ -25,10 +26,25 @@
 
   ; Rows begin empty
   (expect
-    [[] [] [] [] []]
+    (repeat 5 [])
```
Comment (Masclins) [inline backend/test/rules/new_game_test.clj]: I prefere to see what I will get in a test, but ok...

File: backend/test/rules/row_winner_test.clj [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
+    (victory/get-won-rows {:rows [[{:power 1 :owner 5}] [] [] [] []]} 5))
+
   (expect
-    #(= (victory/get-won-rows % 0) 0)
-    (autoplay/as-rules
-      (fn [i] 0)
-      (fn [i] 0))))
+    4
+    (victory/get-won-rows
+      {:rows [
```
Comment (Masclins) [inline backend/test/rules/row_winner_test.clj]: Good move there!

File: backend/test/rules/new_game_test.clj (hunk 1 above)
Comment (kenan-rhoton) [reply backend/test/rules/new_game_test.clj]: This I can see both ways, I changed one direction and back like 6-7 times 😂
I would just ask that we be consistent throughout

**Pull request merged**